### DISCUSSION TITLE
CRUD e Entity Framework > Padronizando o GetById, Dúvida sobre uma das sobrecargas da classe genérica ResultViewModel

### DISCUSSION THREAD

**Discussion by depaula36** [balta-io/forum#558] [Tira Dúvidas] [answered]
Olá, Professor, tudo bem?

Na **aula CRUD e Entity Framework > Padronizando o GetById**, por volta do segundo 0:55 o sr. diz que o tipo da classe genérica a ser gerada indifere (pelo tipo passado em <T>). Compreendi (porque a função da linha 40 - return NotFound(...), além do status code é basicamente passar a msg de erro), mas vendo a sobrecarga dos construtores da classe genérica ResultViewModel, vi que em uma das sobrecargas há uma com parâmetros (T data, List<string> errors). Se instanciasse o objeto passando no catch do método StatusCode(500, new ResultViewModel<Category>(**category**, "Falha..."]) - linha 46 - ao invés de apenas a string, teria acesso ao 'data' até o momento do erro, ou seja, o 'data' incompleto, mas preenchido até o erro (quando esse cenário ocorresse). Seria essa uma das funções dessa sobrecarga? 

Desde já agradeço. Abraço.

<img width="934" alt="image" src="https://github.com/balta-io/forum/assets/78270582/97169adb-de70-4bb9-9529-9c39694de8a1">

**Comment by depaula36** [comment]
<img width="298" alt="image" src="https://github.com/balta-io/forum/assets/78270582/4061eef9-0db6-411e-8b92-c23c1e61c68f">

**Reply by depaula36** [reply]
Perdão, o curso é 'Fundamentos do ASP.NET 6'

**Reply by andrebaltieri** [reply, marked as answer]
Bom dia, @depaula36, isto mesmo, uma das funções do genérico é poder receber diferentes tipos... então no exemplo o `Data` pode ser uma `string`, um `int` ou mesmo um objeto complexo como um `List<Product>`!
